File: todolist-python/app/routes/status_routes.py
from flask import Blueprint, jsonify
import os
import signal
import time
import threading
from app.utils.logger_util import logger

bp = Blueprint('status_routes', __name__)

server_instance = None

def set_server_instance(server):
    global server_instance
    server_instance = server

@bp.route('/check-status', methods=['GET'])
def check_status():
    return jsonify({"success": True, "message": "服务正在运行"})

@bp.route('/shutdown', methods=['POST'])
def shutdown():
    logger.info('收到关闭服务器请求')
    
    def shutdown_server():
        time.sleep(1)
        logger.info('正在关闭服务器...')
        if server_instance:
            server_instance.shutdown()
        else:
            os.kill(os.getpid(), signal.SIGINT)
    
    threading.Thread(target=shutdown_server, daemon=True).start()
    return jsonify({"success": True, "message": "服务器将在1秒后关闭"})

File: todolist-go/app/routes/status_routes.go
package routes

import (
	"net/http"
	"os"
	"syscall"
	"time"

	"github.com/gin-gonic/gin"
)

// RegisterStatusRoutes 注册状态相关路由
func RegisterStatusRoutes(router *gin.Engine) {
	statusGroup := router.Group("/api")
	statusGroup.GET("/check-status", checkStatus)
	statusGroup.POST("/shutdown", shutdown)
}

// checkStatus 健康检查接口
func checkStatus(c *gin.Context) {
	c.JSON(http.StatusOK, map[string]interface{}{
		"success": true, "message": "服务正在运行",
	})
}

// shutdown 关闭服务器接口
func shutdown(c *gin.Context) {
	c.JSON(http.StatusOK, map[string]interface{}{
		"success": true, "message": "服务器将在1秒后关闭",
	})
	go func() {
		time.Sleep(1 * time.Second)
		if p, err := os.FindProcess(os.Getpid()); err == nil {
			p.Signal(syscall.SIGINT)
		}
	}()
}
